filetests/regalloc/gpr-deref-safe-335.clif

```diff
@@ -0,0 +1,44 @@
+test compile
+target x86_64
+
+function u0:587() fast {
+ebb0:
+    v97 = iconst.i32 0
+    v169 = iconst.i32 0
+    v1729 = iconst.i32 0
+    jump ebb100(v97, v97, v97, v97, v97)
+
+ebb100(v1758: i32, v1784: i32, v1845: i32, v1856: i32, v1870: i32):
+    v1762 = iconst.i32 0
+    v1769 = iconst.i32 0
+    v1774 = iconst.i32 0
+    v1864 = iconst.i32 0
+    v1897 = iconst.i32 0
+    jump ebb102(v1774, v1784, v1845, v1856, v1870, v1758, v1762, v169, v1729, v97, v169, v169, v169, v169)
+
+ebb102(v1785: i32, v1789: i32, v1843: i32, v1854: i32, v1868: i32, v1882: i32, v1890: i32, v1901: i32, v1921: i32, v1933: i32, v2058: i32, v2124: i32, v2236: i32, v2366: i32):
+    v1929 = iconst.i32 0
+    v1943 = iconst.i32 0
+    v1949 = iconst.i32 0
+    jump ebb123(v1897, v1769)
+
+ebb123(v1950: i32, v1979: i32):
+    v1955 = iconst.i32 0
+    brz v1955, ebb125
+    jump ebb122(v1929, v1843, v1864, v2058, v1882, v1897, v1943, v1868, v2124, v1901)
+
+ebb125:
+    v1961 = iadd.i32 v1949, v169
+    v1952 = iconst.i32 0
+    v1962 = iconst.i64 0
+    v1963 = load.i32 v1962
+    brz v1963, ebb123(v1952, v1961)
+    jump ebb127
+
+ebb127:
+    v1966 = iconst.i32 0
+    jump ebb122(v1963, v1966, v1966, v1966, v1966, v1966, v1966, v1966, v1966, v1966)
+
+ebb122(v1967: i32, v1971: i32, v1972: i32, v1978: i32, v2032: i32, v2041: i32, v2053: i32, v2076: i32, v2085: i32, v2096: i32):
+    trap user0
+}
```

Comment:
The thing is, the test wasn't crashing if i didn't use `test compile`, so there's something else going on here. I'll investigate a bit more to see what causes this.

Reply:
Also, regarding testing, I've imagined that adding a big real-world test case like one of the wasm demos could be useful to make sure we're not breaking wasm support? Then we could run it on CI, and make sure it doesn't break, or at least warn when it does.

Reply:
> The thing is, the test wasn't crashing if i didn't use test compile, so there's something else going on here. I'll investigate a bit more to see what causes this.

Just looking at the structure of CompileContext::run, I'm pretty sure this difference happens because compile runs optimizations, while regalloc doesn't.

Reply:
Can you dump the IR after the optimizations? Does `test regalloc` reproduce the problem if you run it on that IR?

Reply:
Done in a subsequent push.